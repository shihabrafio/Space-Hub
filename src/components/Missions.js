import '../App.css';
import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchMissions } from '../redux/missions/missionsSlice';
import MissionOne from './Mission';

const Mission = () => {
  const dispatch = useDispatch();
  const { missions, status, error } = useSelector((state) => state.missions);
  useEffect(() => {
    if (status === 'idle') {
      dispatch(fetchMissions());
    }
  }, [status, dispatch]);
  if (status === 'loading') {
    return <h2 style={{ textAlign: 'center' }}>Loading...</h2>;
  }
  if (status === 'failed') {
    return <div>{error}</div>;
  }
  return (
    <table className="mission_table_wrapper">
      <thead>
        <tr className="mission_table_row">
          <th className="mission_table_head">Mission</th>
          <th className="mission_table_head">Description</th>
          <th className="mission_table_head">Status</th>
        </tr>
      </thead>
      <tbody>
        {missions.map((mission) => (
          <tr className="mission_table_rows" key={mission.id}>
            <MissionOne
              name={mission.name}
              description={mission.description}
              id={mission.id}
              status={mission.status}
            />
          </tr>
        ))}
      </tbody>
    </table>
  );
};
export default Mission;
