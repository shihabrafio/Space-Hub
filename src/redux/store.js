import { configureStore } from '@reduxjs/toolkit';
import rocketReducer from './rockets/rocketsSlice';
import missionReducer from './missions/missionsSlice';
import dragonsSlice from './dragons/dragonsSlice';

const store = configureStore({
  reducer: {
    rockets: rocketReducer,
    missions: missionReducer,
    dragons: dragonsSlice,
  },
});
export default store;
