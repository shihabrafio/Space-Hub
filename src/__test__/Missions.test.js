import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { Provider } from 'react-redux';
import Missions from '../components/Missions';
import store from '../redux/store';

const render = (component) => rtlRender(
  <Provider store={store}>
    {component}
  </Provider>,
);

describe('MyRocketss', () => {
  it('render Mission Components', () => {
    render(<Missions />);
    expect(Missions).toMatchSnapshot();
  });
});
