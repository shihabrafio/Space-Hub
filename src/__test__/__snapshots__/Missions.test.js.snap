// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyRocketss render My Rockets Components 1`] = `[Function]`;
